### PR TITLE
Value classes error message improvements

### PR DESCRIPTION
**Pull request by paulp**
This builds on #1326.

### CODE REVIEW

**Comment by jsuereth** [comment]
Wow, lgtm!

**Comment by retronym** [comment]
Could you squash the first two commits, and improve the commit title?

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 1]
```diff
@@ -1659,10 +1670,10 @@ trait Typers extends Modes with Adaptations with Tags {
         _.typedTemplate(cdef.impl, parentTypes(cdef.impl))
       }
       val impl2 = finishMethodSynthesis(impl1, clazz, context)
-      if (clazz.isTrait && clazz.info.parents.nonEmpty && clazz.info.firstParent.normalize.typeSymbol == AnyClass)
+      if (clazz.isTrait && clazz.info.parents.nonEmpty && clazz.info.firstParent.normalize.typeSymbol == AnyClass) {
```
Comment (retronym) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: `.normalize` is redundant here, right?

Comment (paulp) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: Yes, and so is "clazz.info.parents.nonEmpty" for that matter. (If they do happen to be empty, firstParent returns Object.)

**Comment by paulp** [comment]
I still haven't figured out where the line stops between git and github.  To me the "commit title" is the one I checked into git, in this case "Some improvements to value class error messages." I assume you are talking about "Pullreq 1326".  Where and how is that used?

**Comment by paulp** [comment]
Or maybe you mean the title "Fixes SI-6358" from the odersky commits.

**Comment by retronym** [comment]
> Or maybe you mean the title "Fixes SI-6358" from the odersky commits.

Yep, that's what I meant.

**Comment by paulp** [comment]
PLS REBUILD ALL

**Comment by gkossakowski** [comment]
I changed the title of this pull request so it's easier for me (and probably everybody else) to follow what's cooking in pull request queue.

Now we need a reviewer for this one.

**Comment by odersky** [comment]
I think we need to revert edfa41d, so this pull request has to be adapted accordingly.

**Comment by paulp** [comment]
Well, I knew I was gambling basing off a pull request.

**Pull request closed without merging**